pkg/tfcli/client.go

```diff
@@ -202,23 +191,46 @@ func (c *Client) syncPipeline(ctx context.Context, ignoreExitErr bool, command s
 }
 
 // load Terraform state into Client's cache
-func (c *Client) loadStateFromWorkspace() error {
-	pathState := filepath.Join(c.wsPath, fileState)
-	var err error
-	c.tfState, err = c.readFile(pathState)
+func (c *Client) loadStateFromWorkspace() (*json.StateV4, error) {
+	s, err := os.ReadFile(filepath.Join(c.wsPath, fileState))
 	if err != nil {
-		return errors.Wrapf(err, fmtErrLoadState, pathState)
+		return nil, err
 	}
-	return nil
+	st := &json.StateV4{}
+	return st, json.JSParser.Unmarshal(s, st)
 }
 
 func (c *Client) storeStateInWorkspace() error {
-	if c.tfState == nil {
-		return nil
+	attr, err := json.JSParser.Marshal(c.resource.ProduceStateAttributes())
+	if err != nil {
+		return errors.Wrap(err, "cannot marshal produced state attributes")
+	}
+	st := json.NewStateV4()
+	st.TerraformVersion = c.terraformVersion
+	st.Lineage = c.resource.UID
+	st.Resources = []json.ResourceStateV4{
+		{
+			Mode: "managed",
+			Type: c.resource.LabelType,
+			Name: c.resource.LabelName,
+			// TODO(muvaf): we should get the full URL from Dockerfile since
+			// providers don't have to be hosted in registry.terraform.io
+			ProviderConfig: fmt.Sprintf(`provider["registry.terraform.io/%s"]`, c.provider.Source),
+			Instances: []json.InstanceObjectStateV4{
+				{
+					SchemaVersion: 0,
+					PrivateRaw:    []byte(c.resource.PrivateRaw),
+					AttributesRaw: attr,
```

Comment:
We are not building `AttributeSensitivePaths` (`sensitive_attributes`) as part of the state? Are we confident that no resource would require some information from there in its state?

Reply:
No, we are not sure. But I haven't seen a resource put any data there so I don't really know where we can derive it from, its format or what it's used for to understand whether it's another thing to store in annotation.

Reply:
It seems like it's only metadata https://github.com/hashicorp/terraform/issues/25646#issuecomment-718052143 . If it's just what's described here, then it's not really something we need to care about.